doc/Guide to Contributing to Sia.md

```diff
@@ -0,0 +1,257 @@
+* [Contact us](#contact)
+
+<a name="go"/>
+## Get started with Go
+
+<a name="install-go"/>
+### Install Go
+To install Go on your computer, follow the 
+[official installation guide][install-go].  
+
+You should install the latest [official Go binary][binary] for your system (if 
+not available, [install from source][source]).  If you plan to cross compile 
+Sia, see [Cross Compilation with Go 1.5][cross] by Dave Cheney.  
+
+Now make a workspace directory in which you will store source code and 
+dependencies.  You can choose any filepath except where you installed Go (don't 
+choose `/usr/local`).
+
+```bash
+# make a working directory called golang in your home directory
+$ mkdir $HOME/golang
+# store base path in an environmental variable
+$ echo 'export GOPATH=$HOME/golang' >> $HOME/.profile
+# add bin subdirectory to PATH environmental variable
+$ echo 'export PATH=$PATH:$GOPATH/bin' >> $HOME/.profile
+```
+
+Make a base path for your code.  (You'll need [a Github account][signup].)
+```
+$ mkdir -p $GOPATH/src/github.com/<username>
+```
+
+<a name="learn-go"/>
+### Learn Go
+* To get familiar with the language, start with the official [Tour of Go][tour].
+* Move onto [How to Write Go Code][how] to learn how to organize Go packages 
+and use the go tool.
+* Finish with the [Effective Go][effective] guide.
+
+<a name="build-sia"/>
+## Build Sia
+To build Sia on your machine, enter the following on the command line:
+
+```bash
+# Download Sia and its dependencies
+# Binaries will be installed in $GOPATH/bin
+$ go get -u github.com/NebulousLabs/Sia/...
+
+# Switch to directory containing Sia source code
+$ cd $GOPATH/src/github.com/NebulousLabs/Sia
+
+# You have three Sia builds to choose from.
+# To build the standard release binary:
+$ make release-std
+# Or to build the release binary with race detection and an array debugging 
+# asserts:
+$ make release
+# Or to build the developer binary (with a different genesis block, faster 
+# block times, and other changes):
+$ make
+```
+
+<a name="contribute"/>
+## Contribute to the codebase
+<a name="setup-git"/>
+### Set up git
+Install git on your machine according to [these instructions][install-git] in 
+the Pro Git book.
+
+You will first need to set up global settings using the command line.
+```bash
+$ git config --global user.name "Your Name"
+$ git config --global user.email you@somedomain.com
+
+# Tell git to remember your login information for a certain amount of time.
+# Default time is 15 minutes:
+$ git config --global credential.helper cache
+# Or you can choose a different amount of time:
+$ git config --global credential.helper "cache --timeout=[seconds]"
+
+```
+<a name="fork"/>
+### Fork the Sia repository
+While logged into your Github account, navigate to the [Sia repository][sia] 
+and click the 'Fork' button in the upper righthand corner.  Your account now 
+has a 'forked' copy of the original repo at 
+`https://GitHub.com/<your GitHub username>/Sia`.
```

Comment:
I know sorry, thought I caught all of those.  That happened because I got a little trigger happy with text replacement a while back